group/gt.go

```diff
@@ -45,6 +45,23 @@ func (z *GtElement) AddAssign(e Element) Element {
 	return z
 }
 
+func (z *GtElement) Add(a, b Element) Element {
+	z.Set(a)
+	z.AddAssign(b)
+	return z
+}
+
+func (z *GtElement) MultiExp(basis []Element, scalars []fr.Element) (Element, error) {
+	// Maybe quite naive; but it works. Prob could use some Pippenger algorithm?
+	z.inner = bls12381.GT{}
+	for i := 0; i < len(basis); i++ {
+		var tmp GtElement
+		tmp.ScalarMultiplication(basis[i], scalars[i])
+		z.AddAssign(&tmp)
+	}
+	return z, nil
```

Comment:
There's no implementation of `MultiExp` for GT in gnark-crypto.

What I do here is quite naive for simplicity... but as mentioned in the comment, we can try doing a Pippenger/Bucket algorithm. Maybe there's a more clever idea exploiting the fact that this is GT?

In any case, can be optimized in a further phase. Most of the optimizations can mostly focus on making the implementations of this file and forget about the rest of Whisk logic.

Reply:
nice. Let's start with this naive implementation. MSM can be implemented indeed quite efficiently in G_T